trulens/nn/backend/pytorch_backend/pytorch.py

```diff
@@ -45,28 +49,23 @@ def set_default_device(device: str):
     if device is None:
         return old_device
 
-    if isinstance(device, torch.device):
-        device = device.type
+    if isinstance(device, str):
+        device = torch.device(device)
 
-    if "cuda" in device:
-        default_device = torch.device(device)
-    elif device == "cpu":
-        default_device = torch.device(device)
-    else:
-        raise ValueError(f"Unhandled device type {device}")
+    default_device = device
```

Comment:
maybe I'm wrong but I think we need a `nonlocal` or `global` to scope `default_device`